proxy/http/HttpConnectionCount.cc

```diff
@@ -21,51 +21,316 @@
-ConnectionCountQueue ConnectionCountQueue::_connectionCount;
+OutboundConnTrack::Imp OutboundConnTrack::_imp;
+// const std::chrono::seconds OutboundConnTrack::Group::ALERT_DELAY{60};
+
+OutboundConnTrack::GlobalConfig *OutboundConnTrack::_global_config{nullptr};
+
+const MgmtConverter OutboundConnTrack::MAX_CONV{
+  [](void *data) -> MgmtInt { return static_cast<MgmtInt>(*static_cast<decltype(TxnConfig::max) *>(data)); },
+  [](void *data, MgmtInt i) -> void { *static_cast<decltype(TxnConfig::max) *>(data) = static_cast<decltype(TxnConfig::max)>(i); },
+  nullptr,
+  nullptr,
+  nullptr,
+  nullptr};
+
+// Do integer and string conversions.
+const MgmtConverter OutboundConnTrack::MATCH_CONV{
+  [](void *data) -> MgmtInt { return static_cast<MgmtInt>(*static_cast<decltype(TxnConfig::match) *>(data)); },
+  [](void *data, MgmtInt i) -> void {
+    //    *static_cast<decltype(TxnConfig::match) *>(data) = std::clamp(static_cast<decltype(TxnConfig::match)>(i), MATCH_IP,
+    //    MATCH_BOTH);
+    *static_cast<decltype(TxnConfig::match) *>(data) = static_cast<decltype(TxnConfig::match)>(i);
+  },
+  nullptr,
+  nullptr,
+  [](void *data) -> std::string_view {
+    auto t = *static_cast<OutboundConnTrack::MatchType *>(data);
+    return t < 0 || t > OutboundConnTrack::MATCH_BOTH ? "Invalid"sv : OutboundConnTrack::MATCH_TYPE_NAME[t];
+  },
+  [](void *data, std::string_view src) -> void {
+    OutboundConnTrack::MatchType t;
+    if (OutboundConnTrack::lookup_match_type(src, t)) {
+      *static_cast<OutboundConnTrack::MatchType *>(data) = t;
+    } else {
+      OutboundConnTrack::Warning_Bad_Match_Type(src);
+    }
+  }};
+
+// Unfortunately these are not used in RecordsConfig.cc so that must be made consistent by hand.
+const std::string_view OutboundConnTrack::CONFIG_VAR_MAX{"proxy.config.http.per_server.max_connections"};
+const std::string_view OutboundConnTrack::CONFIG_VAR_MATCH{"proxy.config.http.per_server.match"};
+const std::string_view OutboundConnTrack::CONFIG_VAR_QUEUE_SIZE{"proxy.config.http.per_server.queue_size"};
+const std::string_view OutboundConnTrack::CONFIG_VAR_QUEUE_DELAY{"proxy.config.http.per_server.queue_delay"};
+const std::string_view OutboundConnTrack::CONFIG_VAR_ALERT_DELAY{"proxy.config.http.per_server.alert_delay"};
+
+const std::array<std::string_view, static_cast<int>(OutboundConnTrack::MATCH_BOTH) + 1> OutboundConnTrack::MATCH_TYPE_NAME{
+  {"ip"sv, "port"sv, "host"sv, "both"sv}};
+
+// Make sure the clock is millisecond resolution or finer.
+static_assert(OutboundConnTrack::Group::Clock::period::num == 1);
+static_assert(OutboundConnTrack::Group::Clock::period::den >= 1000);
+
+// Configuration callback functions.
+namespace
+{
+int
+Config_Update_Conntrack_Max(const char *name, RecDataT dtype, RecData data, void *cookie)
+{
+  auto config = static_cast<OutboundConnTrack::TxnConfig *>(cookie);
+
+  if (RECD_INT == dtype) {
+    config->max = data.rec_int;
+  }
+  return REC_ERR_OKAY;
+}
```

Comment:
Or, for more limited cut-n-paste action, https://godbolt.org/g/p5qEF3 .

Reply:
I don't think that will handle the time based ones - you can't just assign an integer to a `std::chrono` instance. There are only two bare assigns, not really worth a template for that.

Reply:
Well then just write all the needed overloads for Config_Update_Setter()  https://godbolt.org/g/xRW3ht .  Do I have to do every little thing for you or what?

Reply:
I'm just not sure the benefit is worth the effort.

Reply:
http://www.worldoflongmire.com/features/apes/planet5/PAPERWAD.JPG